src/integrationTest/java/org/opensearch/security/plugin/SystemIndexPlugin1.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.security.plugin;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.node.DiscoveryNodes;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.IndexScopedSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.settings.SettingsFilter;
+import org.opensearch.core.action.ActionResponse;
+import org.opensearch.core.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.env.Environment;
+import org.opensearch.env.NodeEnvironment;
+import org.opensearch.identity.PluginSubject;
+import org.opensearch.indices.SystemIndexDescriptor;
+import org.opensearch.plugins.IdentityAwarePlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.plugins.SystemIndexPlugin;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.rest.RestController;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.script.ScriptService;
+import org.opensearch.security.identity.PluginContextSwitcher;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.watcher.ResourceWatcherService;
+
+public class SystemIndexPlugin1 extends Plugin implements SystemIndexPlugin, IdentityAwarePlugin {
+    public static final String SYSTEM_INDEX_1 = ".system-index1";
+
+    private PluginContextSwitcher contextSwitcher;
+
+    private Client client;
+
+    @Override
+    public Collection<Object> createComponents(
+        Client client,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ResourceWatcherService resourceWatcherService,
+        ScriptService scriptService,
+        NamedXContentRegistry xContentRegistry,
+        Environment environment,
+        NodeEnvironment nodeEnvironment,
+        NamedWriteableRegistry namedWriteableRegistry,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Supplier<RepositoriesService> repositoriesServiceSupplier
+    ) {
+        this.client = client;
+        this.contextSwitcher = new PluginContextSwitcher();
```

Comment:
I am a bit confused by this:

This makes the class `PluginContextSwitcher` (from the security plugin) available for injection. IIRC, there is no isolation between plugins regarding injection, a class provided by plugin A can be injected in plugin B. (Or is there a distinction between modules and plugins?)

Now, if there would be two plugins which make an instance of `PluginContextSwitcher` available for injection, wouldnt lead this to a conflict? 

ATM, I don't think it makes sense to make instances of classes which do not belong to your own plugin available for injection.

Reply:
How would another plugin inject this if they don't have a dependency on the security plugin?

From my understanding, this could be injectable by other plugins but only if they can import this class into their plugin. This can be done by some plugins that supply a SPI (i.e. the Job Scheduler SPI and the LockService), but otherwise if this class is not accessible by other plugins then this class cannot be injected.

Reply:
I see. I was a bit confused by the fact that `PluginContextSwitcher` is located in the `main` sources, but only used by `SystemIndexPlugin1` which I interpreted as a kind of request for re-use. Wouldn't it make sense to move `PluginContextSwitcher` then to the `integrationTest` sources to illustrate that it just belongs to this particular test?

Reply:
Moved this class into the `integrationTest` sourceSet.